src/text-editor.coffee

```diff
@@ -609,15 +609,19 @@ class TextEditor extends Model
   # * "<filename>" when it is the only editing buffer with this file name.
   # * "<filename> — <unique-dir-prefix>" when other buffers have this file name.
   #
+  # * `textEditors` An optional {Array} of the other {TextEditor} instances that
```

Comment:
> Which makes me feel like we should hold off on splitting these two roles apart until the run up to v2.0.

I think I am okay with that and I can see how it would let us move forward here, but it also makes me feel nervous mainly because of two reasons:

1. `textEditors` feels like a _false extension point_. I am not sure package authors (nor outside libraries/software) are actually going to use it.
2. I think we haven't talked about when `v2.0` is going to happen precisely, which could mean we'd have to stick with this new API for a while.

So I think we basically have three alternatives here:

1. Separate `TextEditorPaneItem` and `TextEditor`, dealing with all the corner cases (e.g. what happens if someone calls `getActivePaneItem() instanceof TextEditor`?) , and handling deprecations appropriately.
2. Pass `textEditors` as a parameter. However: avoid documenting it, avoid emitting a deprecation, and default it to `this.workspace.getTextEditors()`, as we know already this API is unstable and it's gonna change at some point in the future.
3. Pass a `getTextEditors` function in the constructor, and default it to `() => []`. When instantiating it through `Workspace`, pass `this.getTextEditors.bind(this)`.

Personally, I'd favor (in order of preference) 1), 3), 2), but I can definitely be talked out of it. @joshaber @lee-dohm: thoughts? 💭